### PR TITLE
Fixed flakes and added parallelization to ChainComponents tests

### PR DESCRIPTION
**Pull request by silaslenihan**
This PR changes the ChainComponents EVM testing interface to support parallel tests, enabling isolated test state and removing potential race conditions. 

**Local verification to confirm the tests are flake free**
Total runs: 500
Failures: 0
Pass rate: 100.00%
Average test time: 3m11.98371491s
Total time: 26h39m51.85745504s

 

### Requires
- https://github.com/smartcontractkit/chainlink-common/pull/965
- https://github.com/smartcontractkit/chainlink-solana/pull/986

### CODE REVIEW

File: core/services/relay/evm/chain_components_test.go [hunk 1]
```diff
@@ -228,17 +230,51 @@ func TestChainReader_HealthReport(t *testing.T) {
...
+		wrapped.Setup(t)
+
+		RunChainComponentsInLoopEvmTests[*testing.T](t, wrapped, true)
+	})
+
+	t.Run("RunChainComponentsInLoopEvmTestsWithBindings", func(t *testing.T) {
+		t.Parallel()
+		it := &EVMChainComponentsInterfaceTester[*testing.T]{Helper: helper, DeployLock: &deployLock}
+		wrapped := WrapContractReaderTesterWithBindings(t, it)
+		// TODO, generated binding tests are broken
```
Comment (jmank88) [inline core/services/relay/evm/chain_components_test.go:272]: TODO for now or later? If later, is there a ticket?

Comment (silaslenihan) [reply core/services/relay/evm/chain_components_test.go:272]: This was already there - not sure if there is a ticket. @ilija42 @pablolagreca

File: core/services/relay/evm/evmtesting/chain_components_interface_tester.go [hunk 2]
```diff
@@ -357,138 +309,108 @@ func (it *EVMChainComponentsInterfaceTester[T]) GetAccountString(i int) string {
...
+
+	cw = it.Helper.WrappedChainWriter(cw, it.client)
 
 	require.NoError(t, err)
 	require.NoError(t, cw.Start(ctx))
-	return it.cw
+
+	t.Cleanup(func() {
+		cw.Close()
+	})
```
Comment (jmank88) [inline core/services/relay/evm/evmtesting/chain_components_interface_tester.go]:
```go
servicetest.Run(t, cw)
```

File: core/services/relay/evm/chain_components_test.go [hunk 3]
```diff
@@ -228,17 +230,51 @@ func TestChainReader_HealthReport(t *testing.T) {
...
+	helper.Init(t)
+	deployLock := sync.Mutex{}
 	// add new subtests here so that it can be run on real chains too
-	RunChainComponentsEvmTests(t, it)
-	RunChainComponentsInLoopEvmTests[*testing.T](t, commontestutils.WrapContractReaderTesterForLoop(it))
-	RunChainComponentsInLoopEvmTests(t, WrapContractReaderTesterWithBindings(t, it))
+	t.Run("RunChainComponentsEvmTests", func(t *testing.T) {
+		t.Parallel()
+		it := &EVMChainComponentsInterfaceTester[*testing.T]{Helper: helper, DeployLock: &deployLock}
+		// These tests are broken in develop as well, so disable them for now
```
Comment (ilija42) [inline core/services/relay/evm/chain_components_test.go:242]: CC @poopoothegorilla  Can you create an epic for general Chain Components work that is not urgent, but has to be done? Maybe just have 1 for now and then have them per chain

**Review by ilija42 — DISMISSED** [review]
Good job!

File: core/services/relay/evm/chain_components_test.go (hunk 3 above)
Comment (poopoothegorilla) [reply core/services/relay/evm/chain_components_test.go:242]: Created this epic https://smartcontract-it.atlassian.net/browse/NONEVM-1084 used the word "Hardening" which some other projects use for this type of work

**Review by ilija42 — DISMISSED** [review]

**Review by Farber98 — DISMISSED** [review]
great job with this! 💪

**Review by Farber98 — DISMISSED** [review]

**Review by ilija42 — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by ilija42 — APPROVED** [review]

**Pull request merged**